packages/svelte/messages/compile-warnings/template.md

```diff
@@ -38,6 +38,18 @@
 
 > Using `on:%name%` to listen to the %name% event is deprecated. Use the event attribute `on%name%` instead
 
+## node_invalid_placement_ssr
+
+> %thing% is invalid inside <%parent%>. When rendering this component on the server, the resulting HTML will be modified by the browser, likely resulting in a `hydration_mismatch` warning
+
+HTML has some restrictions where certain elements can appear. For example, a `<div>` inside a `<p>` is invalid. Some violations "only" result in invalid HTML, others result in the HTML being repaired by the browser, resulting in content shifting around. Some examples:
+
+- `<p>hello <div>world</div></p>` will result in `<p>hello </p><div>world</div><p></p>` for example (the `<div>` autoclosed the `<p>`)
+- `<option><div>option a</div></option>` will result in `<option>option a</option>` (the `<div>` is removed)
+- `<table><tr><td>cell</td></tr></table>` will result in `<table><tbody><tr><td>cell</td></tr></tbody></table>` (a `<tbody>` is auto-inserted)
+
+Svelte issues a compiler warning when it detects that it will generate the HTML in such a way that it will work on the client, but always fail when using server side rendering, because the resulting HTML will be repaired and result in the client runtime not finding the nodes at the expected locations when hydrating the DOM.
```

Comment:
We're in the compiler warnings section of the (hypothetical future) docs, I don't think we need to say that Svelte will issue a compiler warning — if someone is here it's most likely because they just experienced one

```suggestion
HTML restricts where certain elements can appear. For example, a `<p>` element cannot contain another `<p>`, or a `<div>`, or any other block-level element. The browser will 'repair' the HTML in a way that breaks Svelte's assumptions about the structure of your components. Some more examples:

- `<option><div>...</div></option>` will result in `<option>...</option>` (everything except text is removed)
- `<table><tr><td>...</td></tr></table>` will result in `<table><tbody><tr><td>...</td></tr></tbody></table>` (a `<tbody>` is auto-inserted)

This code will work when the component is rendered on the client, but if you use server rendering it will cause hydration to fail.
```

Reply:
The idea was that there's a compiler error version of this and so there's a bit of context why there's also a warning version of this (thought that's likely not very clear still, and so we can either just omit it [your suggestion] or expand on it by actually referencing the other variant)

Reply:
what about 'This code will work when the component is rendered on the client (which is why this is a warning rather than an error), but if you use server rendering it will cause hydration to fail.'

Reply:
yeah that sounds good. My suggestion then (mirroring my other error text suggestion):
```suggestion
HTML restricts where certain elements can appear. In case of a violation the browser will 'repair' the HTML in a way that breaks Svelte's assumptions about the structure of your components. Some examples:

- `<p>hello <div>world</div></p>` will result in `<p>hello </p><div>world</div><p></p>` for example (the `<div>` autoclosed the `<p>` because `<p>` cannot contain block-level elements)
- `<option><div>option a</div></option>` will result in `<option>option a</option>` (the `<div>` is removed)
- `<table><tr><td>cell</td></tr></table>` will result in `<table><tbody><tr><td>cell</td></tr></tbody></table>` (a `<tbody>` is auto-inserted)

This code will work when the component is rendered on the client (which is why this is a warning rather than an error), but if you use server rendering it will cause hydration to fail.
```